src/main/java/net/mcreator/workspace/elements/ModElement.java

```diff
@@ -324,4 +323,26 @@ public void setParentFolder(@Nullable FolderElement parent) {
 			this.path = parent.getPath();
 	}
 
+	public static class ModElementDeserializer implements JsonDeserializer<ModElement> {
+
+		private final Gson gson = new Gson();
+
+		@Override
+		public ModElement deserialize(JsonElement jsonElement, Type typeOfT, JsonDeserializationContext context)
+				throws JsonParseException {
+			JsonObject json = jsonElement.getAsJsonObject();
+
+			String newType = json.get("type").getAsString();
+			if (newType.equals("gun")) {
+				newType = "rangeditem";
+			} else if (newType.equals("mob")) {
+				newType = "livingentity";
+			}
+
+			json.addProperty("type", newType);
+
+			return gson.fromJson(json, ModElement.class);

```

Comment:
If the errors are caused by using `addProperty()` here, I'd suggest adding a private method to the `ModElement` class that returns its type as a string (e. g. `getTypeString()`) and change deserializing code like this:
```suggestion
			ModElement retVal = gson.fromJson(json, ModElement.class);

			if (retVal.getTypeString().equals("gun")) {
				retVal.setType(ModElementType.RANGEDITEM);
			} else if (retVal.getTypeString().equals("mob")) {
				retVal.setType(ModElementType.LIVINGENTITY);
			}

			return retVal;
```

Reply:
Previous approach is fine.